targets/galatea/base.py

```diff
@@ -11,217 +9,57 @@
-        self.specials += Instance("ODDR2", p_DDR_ALIGNMENT="NONE",
-                                  p_INIT=0, p_SRTYPE="SYNC",
-                                  i_D0=1, i_D1=0, i_S=0, i_R=0, i_CE=1,
-                                  i_C0=clk_sdram_half_shifted,
-                                  i_C1=~clk_sdram_half_shifted,
-                                  o_Q=output_clk)
-        self.specials += Instance("OBUFDS", i_I=output_clk, o_O=clk.p, o_OB=clk.n)
-
-        # Peripheral clock - 50MHz
-        # ------------------------------------------------------------------------------
-        # The peripheral clock is kept separate from the system clock to allow
-        # the system clock to be increased in the future.
-        dcm_base50_locked = Signal()
-        self.specials += [
-            Instance("DCM_CLKGEN", name="crg_periph_dcm_clkgen",
-                     p_CLKIN_PERIOD=10.0,
-                     p_CLKFX_MULTIPLY=2,
-                     p_CLKFX_DIVIDE=4,
-                     p_CLKFX_MD_MAX=0.5, # CLKFX_MULTIPLY/CLKFX_DIVIDE
-                     p_CLKFXDV_DIVIDE=2,
-                     p_SPREAD_SPECTRUM="NONE",
-                     p_STARTUP_WAIT="FALSE",
-
-                     i_CLKIN=clk100a,
-                     o_CLKFX=self.cd_base50.clk,
-                     o_LOCKED=dcm_base50_locked,
-                     i_FREEZEDCM=0,
-                     i_RST=ResetSignal(),
-                     ),
-            AsyncResetSynchronizer(self.cd_base50,
-                self.cd_sys.rst | ~dcm_base50_locked)
-        ]
-        platform.add_period_constraint(self.cd_base50.clk, 20)
+from targets.utils import dict_set_max
+from .crg import _CRG
 
 
 class BaseSoC(SoCSDRAM):
-    csr_peripherals = (
-        "spiflash",
-        "ddrphy",
-        "info",
-    )
-    csr_map_update(SoCSDRAM.csr_map, csr_peripherals)
-
-    # FIXME: Add spiflash
-    #mem_map = {
-    #    "spiflash": 0x20000000,  # (default shadow @0xa0000000)
-    #}
-    #mem_map.update(SoCSDRAM.mem_map)
-
-    def __init__(self, platform, **kwargs):
-        if 'integrated_rom_size' not in kwargs:
-            kwargs['integrated_rom_size']=0x8000
-        if 'integrated_sram_size' not in kwargs:
-            kwargs['integrated_sram_size']=0x4000
-
-        if 'expansion' in kwargs:
-            tofe_board_name = kwargs.get('expansion')
-            del kwargs['expansion']
-        else:
-            tofe_board_name = None
-
-
-        clk_freq = 50*1000000
-        SoCSDRAM.__init__(self, platform, clk_freq, **kwargs)
-
-        self.submodules.crg = _CRG(platform, clk_freq)
-        self.platform.add_period_constraint(self.crg.cd_sys.clk, 1e9/clk_freq)
-
-        # Basic peripherals
+    mem_map = {**SoCSDRAM.mem_map, **{
+        'spiflash': 0x20000000,
+    }}
+
+    def __init__(self, platform, spiflash="spiflash_1x", **kwargs):
+        dict_set_max(kwargs, 'integrated_rom_size', 0x8000)
+        dict_set_max(kwargs, 'integrated_sram_size', 0x4000)
```

Comment:
@mithro - https://github.com/timvideos/litex-buildenv/pull/311/files#diff-cbdf402788c4a014f5ad1b7253b4f18eR26 changes integrated sram size for this platform to `0x8000` - is this intended or just a typo?

Reply:
Does lines 188-192 on the left indicate that was the original sizes?

Reply:
On the current master `integrated_sram_size` for `galatea` target is set to 0x4000 - https://github.com/timvideos/litex-buildenv/blob/master/targets/galatea/base.py#L192.